### ISSUE TITLE
[BUG] Creating storage account Premium

### ISSUE THREAD

**Issue by vladciobancai** [claranet/terraform-azurerm-storage-account#12]
### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.5.7

### AzureRM Provider Version

3.85.0

### Affected Resource(s)/Data Source(s)

azurerm_storage_account

### Terraform Configuration Files

```hcl
module "app_static_web_auth_test" {
  source  = "claranet/storage-account/azurerm"
  version = "7.8.0"

  storage_account_custom_name = "test${terraform.workspace}staticweb"
  client_name                 = "static-web-test"
  environment                 = local.environment
  location                    = data.terraform_remote_state.core-infra.outputs.azure_location
  location_short              = data.terraform_remote_state.core-infra.outputs.azure_short_location
  resource_group_name         = data.terraform_remote_state.core-infra.outputs.azure_resource_group_name
  stack                       = "auth-test"
  default_tags_enabled        = false
  extra_tags                  = local.tags
  https_traffic_only_enabled  = true
  account_tier                = "Premium"
  account_kind                = "BlockBlobStorage"
  access_tier                 = "Hot"
  account_replication_type    = "ZRS"

  logs_destinations_ids = []

  storage_blob_cors_rule = {
    allowed_headers    = ["*"]
    allowed_methods    = ["GET", "HEAD"]
    allowed_origins    = ["*"]
    exposed_headers    = ["*"]
    max_age_in_seconds = 3600
  }

  storage_blob_data_protection = {
    change_feed_enabled                       = true
    versioning_enabled                        = true
    delete_retention_policy_in_days           = 30
    container_delete_retention_policy_in_days = 30
    container_point_in_time_restore           = true
  }

  default_firewall_action = "Allow"
  network_rules_enabled   = true
  allowed_cidrs           = ["147.243.0.0/16"]
  network_bypass          = ["AzureServices"]

  static_website_config = {
    index_document     = "benefits.html"
    error_404_document = "error.html"
  }
}
```

### Debug Output/Panic Output

```shell
module.app_static_web_auth_test.azurerm_storage_account.storage: Still creating... [8m20s elapsed]
╷
│ Error: updating `blob_properties`: storage.BlobServicesClient#SetServiceProperties: Failure responding to request: StatusCode=400 -- Original Error: autorest/azure: Service returned an error. Status=400 Code="OperationNotAllowedOnKind" Message="The operation is not allowed on account kind BlockBlobStorage"
│
│   with module.app_static_web_auth_test.azurerm_storage_account.storage,
│   on .terraform/modules/app_static_web_auth_test/r-storage-account.tf line 1, in resource "azurerm_storage_account" "storage":
│    1: resource "azurerm_storage_account" "storage" {
│
```

### Expected Behaviour

Hi, I'm trying to create a storage account as Premium with kind of BlockBlobStorage for a static website and I'm getting the error provided in the output. I'm missing something, could somebody help me ?

Thank you

**Comment by jmapro** [comment]
Hi !

Point In Time restore is not available on Premium Storage accounts. https://learn.microsoft.com/en-us/azure/storage/blobs/point-in-time-restore-overview

> Only block blobs in a standard general-purpose v2 storage account can be restored as part of a point-in-time restore operation. Append blobs, page blobs, and premium block blobs aren't restored.

Try to disable `container_point_in_time_restore`

**Comment by vladciobancai** [comment]
Hi jmapro, worked . It is possible in the README this value to be mentioned if the PREMIUM storage account is configured ?

**Comment by jmapro** [comment]
Hello @vladciobancai, I've created an internal task to add a precondition to deny with a message this type of configuration.

**Comment by jmapro** [comment]
@vladciobancai we've released v7.10.0 which includes a precondition to have a message when you configure PITR on Premium account.
